lib/internal/process/task_queues.js

```diff
@@ -70,7 +71,7 @@ function processTicksAndRejections() {
   let tock;
   do {
     while (tock = queue.shift()) {
-      const asyncId = tock[async_id_symbol];
+      const asyncId = tickId = tock[async_id_symbol];
```

Comment:
This is not exactly correct since it will only run if there is a `nextTick` scheduled and won't take other possible async entry points into account such as IO and what not. This is not my area of expertise. 

Maybe would require some help from @addaleax if this approach is deemed worth looking further into.

Reply:
Yeah, I’m happy to investigate this in more detail… It looks like you’re essentially going for an id for each individual synchronous block of JS execution, which should be doable, but I agree that it makes sense to get a green light for this approach first :)

Reply:
btw, fwiw I've been mulling over the details of a proposal to TC-39 that does precisely that: assign execution and trigger IDs for every JS execution at the language level. It would make several things significantly easier.

I'm not convinced that *this* PR is the way to go forward tho and definitely need to give it a bit more thought.

Reply:
Hm yeah … I’m not sure if this would work fully when using `async_hooks` because we’d basically always want the lowest async id on the stack, not the highest. But maybe we could expose + use that? That would also solve the readable-stream issue for all supported versions of Node.js (although obviously not browsers…)